File: Contadordecafe/app/src/main/java/com/esampaio/apps/contadordecafe/persistence/RepositorioCafe.java
package com.esampaio.apps.contadordecafe.persistence;

import android.app.Activity;
import android.content.Context;
import android.content.SharedPreferences;
import android.content.pm.ActivityInfo;


import java.text.SimpleDateFormat;
import java.util.Date;

/**
 * Created by eduardo on 20/03/17.
 */

public class RepositorioCafe {
    private CafeDAO cafeDAO;

    public RepositorioCafe(Activity activity){
        cafeDAO = new CafeDAO(activity);
    }

    public Integer getQtdCafeHj(){
        return getQtdCafeDia(new Date(System.currentTimeMillis()));
    }
    public void beberCafeHj(){
        cafeDAO.beberCafe();
    }

    public Integer getQtdCafeDia(Date date){
        return cafeDAO.getQtdCafesBebidos(date);
    }


}

File: Contadordecafe/app/src/main/java/com/esampaio/apps/contadordecafe/persistence/RepositorioCafe.kt
package com.esampaio.apps.contadordecafe.persistence

import android.content.Context
import java.util.Date

/**
 * Created by eduardo on 20/03/17.
 */

class RepositorioCafe(context: Context) {
    private val cafeDAO: CafeDAO

    init {
        cafeDAO = CafeDAO(context)
    }

    val qtdCafeHj: Int
        get() = getQtdCafeDia(Date(System.currentTimeMillis()))

    fun beberCafeHj() {
        cafeDAO.beberCafe()
    }

    fun getQtdCafeDia(date: Date): Int {
        return cafeDAO.getQtdCafesBebidos(date)
    }

    fun getQtdCafeDia(initDate: Date,finalDate:Date): Int {
        return cafeDAO.getQtdCafesBebidos(initDate,finalDate);
    }
}
